webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/ldap/LDAPRuleRepository.scala

```diff
@@ -169,6 +169,27 @@ class WoLDAPRuleRepository(
     }
   }
 
+  def deleteSystemRule(id:RuleId, modId: ModificationId, actor:EventActor, reason:Option[String]) : IOResult[DeleteRuleDiff] = {
```

Comment:
You should not duplicate code.

you can either:
* Add a boolean parameter to deleteRule to allow deletion of systemRules
* Do like groups add a private function to internalDelete with a boolean parameter to allow systemRules deletion and two functions deleteRule and deleteSystemRules

I would go for the first one, either to understand to me, but maybe @fanf has another preference on this, but don( duplicate code ;)

Reply:
I prefer to expose different APIs. 

But actually, I'm thinking these methods shoud just go in the plugin (and so duplicate code :) ). 
If we don't have them so far in rudder, I'm almost sure their use case it extremely narrow, and perhaps even limited to that plugin. So I would just enterily factor them out in the plugin, where they are used. It would make evolution (for both rudder and the plugin) easier.